app/Scripts/boid.gd

```diff
@@ -71,7 +103,7 @@ func find_all_boids():
 func cohesion():
 	var percieved_center = Vector2(0,0)
 	for b in boids:
-		if(b!=self):
+		if(b!=self && boids.has(boids.find(b)) ):
```

Comment:
what is `boids.has(boids.find(b)` doing? We're looping through boids, so why isn't `b` already known to be in `boids`?

Reply:
Yep Idk why that happens and for some reason I need to check to see if the boid exist or not, If I don't the script will try to access a dead boid for some reason and gives me errors

Reply:
Can we figure out why?